elide-core/src/main/java/com/yahoo/elide/security/executors/ActivePermissionExecutor.java

```diff
@@ -0,0 +1,219 @@
+/*
+ * Copyright 2016, Yahoo Inc.
+ * Licensed under the Apache License, Version 2.0
+ * See LICENSE file in project root for terms.
+ */
+package com.yahoo.elide.security.executors;
+
+import com.yahoo.elide.core.RequestScope;
+import com.yahoo.elide.core.exceptions.ForbiddenAccessException;
+import com.yahoo.elide.security.ChangeSpec;
+import com.yahoo.elide.security.PermissionExecutor;
+import com.yahoo.elide.security.PersistentResource;
+import com.yahoo.elide.security.SecurityMode;
+import com.yahoo.elide.security.checks.Check;
+import com.yahoo.elide.security.permissions.ExpressionBuilder;
+import com.yahoo.elide.security.permissions.ExpressionResult;
+import com.yahoo.elide.security.permissions.expressions.Expression;
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+
+import java.lang.annotation.Annotation;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Queue;
+import java.util.concurrent.LinkedBlockingQueue;
+
+import static com.yahoo.elide.security.permissions.ExpressionResult.Status.DEFERRED;
+import static com.yahoo.elide.security.permissions.ExpressionResult.Status.FAIL;
+
+/**
+ * Default permission executor.
+ * This executor executes all security checks as outlined in the documentation.
+ */
+public class ActivePermissionExecutor implements PermissionExecutor {
+    private final Queue<QueuedCheck> commitCheckQueue = new LinkedBlockingQueue<>();
+
+    private final RequestScope requestScope;
+    private final ExpressionBuilder expressionBuilder;
+
+    /**
+     * Constructor.
+     *
+     * @param requestScope Request scope.
+     */
+    public ActivePermissionExecutor(final com.yahoo.elide.core.RequestScope requestScope) {
+        HashMap<Class<? extends Check>, Map<PersistentResource, ExpressionResult>> cache = new HashMap<>();
+
+        this.requestScope = requestScope;
```

Comment:
Talked to @wcekan offline. He suggests, instead, we leave this as-is and use `Supplier<PermissionExecutor>` elsewhere.